lumicks/pylake/detail/image.py

```diff
@@ -129,7 +129,40 @@ def seek_timestamp_next_line(infowave):
     return pixel_start[idx + 1]
 
 
-def reconstruct_image_sum(data, infowave, pixels_per_line, lines_per_frame=None):
+def process_infowave(infowave):
+    # Example infowave:
+    #  1 0 0 1 0 1 2 0 1 0 0 1 0 1 0 1 0 0 1 0 2 0 1 0 1 0 0 1 0 1 0 1 2 1 0 0 1
+    #              ^ <-----------------------> ^                       ^
+    #                       one pixel
+    mask = infowave != InfowaveCode.discard
+    subset = infowave[mask]
+
+    # After discard:
+    #  1 1 1 2 1 1 1 1 1 2 1 1 1 1 1 2 1 1 1
+    #        ^ <-------> ^           ^
+    #         pixel_size (i.e. data samples per pixel)
+    # This should be:
+    #   pixel_sizes = np.diff(np.flatnonzero(infowave == InfowaveCode.pixel_boundary))
+    # But for now we assume that every pixel consists of the same number of samples
+    return subset, mask
+
+
+def round_up(size, n):
+        """Round up `size` to the nearest multiple of `n`"""
+        return int(math.ceil(size / n)) * n
+
+
+def reshape_reconstructed_image(pixels, shape):
+    # shape = (<<lines_per_frame>>, pixels_per_line)
+    resized_pixels = np.zeros(round_up(pixels.size, np.prod(shape)))
+    resized_pixels[:pixels.size] = pixels
+    if len(shape) > 1:
+        return resized_pixels.reshape(-1, *shape).squeeze()
+    else:
+        return resized_pixels.reshape(-1, *shape)
```

Comment:
Could an unconditional `.squeeze()` work here? I don't exactly remember why it would not. It would be good to either always do `.squeeze()` or add a comment indicating why there's a difference.

Reply:
From the tests it seems like the only time `squeeze` should not be called is for a single-line Kymo (data expected to always be >= 2D) so I moved that check to the `_to_spatial()` method in the Kymo class. I think it's a lot cleaner this way

Reply:
I wonder if we can get something more controlled than squeeze because now it's kind of offloading dimensionality checks to the end user.

If you have a 2D scan with one line, now, you suddenly get 1D data instead of 2D. If you have a 3D scan with only one plane, you get a 2D array. I don't like that the dimensionality turns dynamic depending on how much data you acquired (rather than the expected dimensionality from shape). It would make code using the result from reconstructions having to check the dimensionality.

Reply:
That's a good point, Joep. Maybe it would be best to flip it: don't squeeze by default in `reshape_reconstructed_image()`. Only do it in the end class if it makes sense.